modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/util/OpenIdConnectTokenRequestUtil.java

```diff
@@ -101,12 +101,20 @@ private static OIDCTokens _requestOIDCTokens(
 
 		URI uri = oidcProviderMetadata.getTokenEndpointURI();
 
-		ClientID clientID = new ClientID(openIdConnectProvider.getClientId());
-		Secret secret = new Secret(openIdConnectProvider.getClientSecret());
```

Comment:
@arthurchan35 is it ok to couple `OpenIdConnectTokenRequestUtil` to our `OpenIdConnectProviderImpl`?  The instance is obtained from `OpenIdConnectProviderRegistry` which is a SPI.

We actually used to have an unsafe cast to `OpenIdConnectProviderImpl` , but this was removed with our recent changes for the refresh token impl.